overlord/ifacestate/handlers.go

```diff
@@ -540,6 +543,26 @@ func (m *InterfaceManager) doAutoConnect(task *state.Task, _ *tomb.Tomb) error {
 
 	chg := task.Change()
 
+	// content iface has special handling for the default-provider
+	// auto-install
+	for _, plug := range m.repo.Plugs(snapName) {
+		if plug.Interface == "content" {
+			if s, _ := plug.Attrs["content"].(string); s != "" {
+				if dprovider, _ := plug.Attrs["default-provider"].(string); dprovider != "" {
+					for _, t := range chg.Tasks() {
```

Comment:
not sure why we are checking only this change and all changes?

all probably best to have an helper that gives us a list of default-providers?

and to the go over tasks once